solution/index.js

```diff
@@ -0,0 +1,212 @@
+    loaderElement.id = loaderElementId;
+    const addLoaderToElement = isLoad ? loadButton : saveButton;
+    addLoaderToElement.append(loaderElement);
+}
+
+function removeLodingElement(isLoad){
+    const loaderElementId = isLoad ? "load-loader" : "save-loader";
+    const loaderElement = document.getElementById(loaderElementId);
+    loaderElement.remove(); 
+}
+
+async function loadDataFromApi(){
+    addLodingElement(true);
+    const downlodingResponse = await fetch("https://json-bins.herokuapp.com/bin/614b1a584021ac0e6c080cea", {
+        method : "GET"
+    });
+    const response = await downlodingResponse.json();
+    const dataObj =  response.tasks;
+    if (downlodingResponse.status != 200 ){
+        alert ("An error has occured when trying to send the data to the server. error status: "
+            + downlodingResponse.status + " error message: " + downlodingResponse.statusText);
+        return
+    }
+    localStorage.setItem("tasks", JSON.stringify(dataObj));
+    clearAllTasks()
+    reloadPage()
+    removeLodingElement(true);
+}
+
+async function saveDataToApi(){
+    addLodingElement(false);
+    const uploadResponse = await fetch("https://json-bins.herokuapp.com/bin/614b1a584021ac0e6c080cea",{
+        method : "PUT",
+        headers: {
+            "Content-Type": "application/json",
+            "Accept": "application/json",
+        },
+        body: JSON.stringify({ tasks: JSON.parse(localStorage.getItem("tasks"))})
+    });
+    if (uploadResponse.status != 200 ){
+        alert ("An error has occured when trying to send the data to the server. error status: "
+            + uploadResponse.status + " error message: " + uploadResponse.statusText);
+        return
+    }
+    removeLodingElement(false);
+}
+
+function mapIndexToStatus(index) {
+    switch(index){
+        case 0: return "todo";
+        case 1: return "in-progress";
+        case 2: return "done";
+    }  
+}
+
+function createListElement(list, inputValue) {
+    const ulElementsList = document.getElementsByClassName("list");
+    let newLi = document.createElement("LI");
+    newLi.classList.add("task");
+    newLi.innerText = inputValue;  
+
+    newLi.addEventListener("mouseover", function() {
+        document.addEventListener("keydown", whileHoverAndKeyPress);
+    });
+    newLi.addEventListener("mouseout", function() {
+        document.removeEventListener("keydown", whileHoverAndKeyPress);
+    })
+    
+    newLi.addEventListener("dblclick", function(){    // make editable by dblclick
+        newLi.setAttribute("contenteditable", "true");
+    }); 
+    newLi.addEventListener("blur", function(){   
+        const currentStatus = getCurrentStatus(ulElementsList, newLi.parentNode);
+        updateLocalStorage(inputValue, currentStatus, newLi.innerText);
+        newLi.setAttribute("contenteditable", "false");
+    }); 
+
+    list.prepend(newLi); 
+
+    function whileHoverAndKeyPress(keypressEvent) {
```

Comment:
Nice implementation. Can you think of a way to put this function outside `createListElement`?